starknet_py/contract.py

```diff
@@ -398,44 +408,39 @@ async def from_address(
         should create Contract's instances directly instead of using this function to avoid unnecessary API calls.
 
         :raises ContractNotFoundError: when contract is not found
-        :raises TypeError: when Client not supporting `get_code` methods is used
+        :raises TypeError: when given client's `get_class_by_hash` method does not return abi
+        :raises ProxyResolutionError: when given ProxyChecks were not sufficient to resolve proxy's implementation
         :param address: Contract's address
         :param client: Client
         :param proxy_config: Proxy resolving config
-            If set to ``True``, will use default proxy checks and :class:
+            If set to ``True``, will use default proxy checks :class:
             `starknet_py.proxy_check.OpenZeppelinProxyCheck`
-            and :class:`starknet_py.proxy_check.ArgentProxyCheck` and default max_steps = 5.
+            and :class:`starknet_py.proxy_check.ArgentProxyCheck`.
 
             If set to ``False``, :meth:`Contract.from_address` will not resolve proxies.
 
-            If a valid `ProxyConfig` is provided, will use values from that instead supplementing
+            If a valid `ProxyConfig` is provided, will use its values instead, supplementing
             with defaults when needed.
 
         :return: an initialized Contract instance
         """
-        default_config: Contract.ProxyConfig = {
-            "max_steps": 5,
-            "proxy_checks": [ArgentProxyCheck(), OpenZeppelinProxyCheck()],
-        }
-        if isinstance(proxy_config, bool):
-            proxy_config = default_config if proxy_config else {}
-        else:
-            proxy_config = {**default_config, **proxy_config}
-
-        proxy_client = client.client if isinstance(client, AccountClient) else client
-        if not isinstance(proxy_client, GatewayClient):
+        address = parse_address(address)
+        proxy_config = prepare_proxy_config(proxy_config)
+        actual_client = get_actual_client(client)
+        if not isinstance(actual_client, GatewayClient):
```

Comment:
Can't we support any client now that we use contract classes?

Reply:
There is no way of getting `abi` from a contract using RPC v0.1.0, so only GatewayClient can be used for now.